init.c

```diff
@@ -934,5 +941,29 @@ int luaopen_libcutorch(lua_State *L)
   lua_pushlightuserdata(L, state);
   lua_setfield(L, -2, "_state");
 
+  /* when cutorch goes out of scope, we need to make sure THCState is properly 
+     shut down (so that memory doesn not leak. Since _state is a lightuserdata
+     we cannot associate an __gc method with it. Hence, create a userdata, and
+     associate a metatable with it, which has an __gc method which properly 
```

Comment:
I thought about this, but that would mean lots of changes to all packages that use cutorch state.